pkg/cliplugins/workspace/plugin/kubeconfig.go

```diff
@@ -232,26 +232,11 @@ func (kc *KubeConfig) UseWorkspace(ctx context.Context, opts *Options, workspace
 	workspaceConfigCurrentContext := workspaceConfig.CurrentContext
 	workspaceContextName := kcpWorkspaceContextNamePrefix + workspaceConfigCurrentContext
 
-	currentContextName := kc.startingConfig.CurrentContext
-	if opts.KubectlOverrides.CurrentContext != "" {
-		currentContextName = opts.KubectlOverrides.CurrentContext
-	}
-
-	currentContext := kc.startingConfig.Contexts[currentContextName]
-	var currentContextAuthInfo *api.AuthInfo
-	if currentContext != nil {
-		currentContextAuthInfo = kc.startingConfig.AuthInfos[currentContext.AuthInfo]
-	}
 	kc.startingConfig.Clusters[workspaceContextName] = workspaceConfig.Clusters[workspaceConfigCurrentContext]
 
-	kubectlOverrides := opts.KubectlOverrides
-
-	workspaceAuthInfo := prioritizedAuthInfo(&kubectlOverrides.AuthInfo, currentContextAuthInfo)
-	kc.startingConfig.AuthInfos[workspaceContextName] = workspaceAuthInfo
-
 	kc.startingConfig.Contexts[workspaceContextName] = &api.Context{
 		Cluster:  workspaceContextName,
-		AuthInfo: workspaceContextName,
+		AuthInfo: kc.startingConfig.Contexts[kc.startingConfig.CurrentContext].AuthInfo,
```

Comment:
Seems like you completely disable to use of `kubectl` overrides. Which means that you cannot do something like:
`kubectl kcp workspace --token=user-1-token use workspace1` to override AuthInfo to be the `user1` when accessing `workspace1`.

Is it really what we want ?

Reply:
Probably not.  I'll revise.

Reply:
I'm refactoring the whole piece of code, so that I have to touch this part as well.
Is it OK if I include the requirement you started answering in this PR ?

Reply:
Asking people to wait on small PRs because enormous ones are coming is IMHO an anti-pattern - not only because enormous PRs are not great to write and review, but because no contributor likes to hear that their work will be done by someone else / that they need to wait.

Reply:
I updated it, but no worries if you refactor it.